### PR TITLE
Support to generate xcode project files via CMake. 

### PR DESCRIPTION
**Pull request by fosterbrereton**
Note the build directory is a _sibling_ to adobe_source_libraries, and that the debug and release builds are separate projects.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,26 +7,149 @@ project(adobe_source_libraries CXX)
 
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LANGUAGE_STANDARD "c++0x")
+set(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LIBRARY "libc++")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fconstexpr-depth=1024 -ftemplate-depth=1024")
```
Comment (ilelann) [inline CMakeLists.txt]: simple curiosity :
target_compile_options(asl PUBLIC $<$<CXX_COMPILER_ID:Clang>:-fconstexpr-depth=1024>)
lines in source/CMakeLists.txt do not handle this properly ?

Comment (fosterbrereton) [inline CMakeLists.txt]: That's a great question, @ilelann. All I can tell you is Xcode wouldn't build something without the flag added where I put it. I didn't see it in the compiler setting when building one of the test apps (I think it was the FNV one.) I'm not well versed in CMake, so it's possible I've missed something. If I have, let me know.

**Comment by fosterbrereton** [comment]
@ilelann I figured out what it was: The Apple compiler is called `AppleClang`, so wasn't including the flags. I have modified the expression and everything seems to be building as expected. Unfortunately I couldn't figure out the generator expression syntax, and fell back on something I could get working. If you'd like to improve upon it, feel free.

**Comment by ilelann** [comment]
@fosterbrereton I'm considering changing external project handling. 
I pushed with ExternalProject but don't really like the resulting behavior.
ExternalProject is doing everything at build time, which is cool but has limitations in terms of "seamless" import : you clone for each build configuration and you don't get the imported project source files into IDE.
I'm considering something like this pseudo code
```cmake
if (not use packaged foo)
    if (not cloned foo) 
        clone foo ..
    endif()
    add_subirectory(../foo ${CMAKE_BINARY_DIR}/imported/foo)
else()
    find_package(foo)
endif()
```
It should be closer to configure.sh behavior. I'm testing this to import ASL into APL.
If considered OK, I might generalize to double-conversion, and maybe even boost.

**Comment by fosterbrereton** [comment]
@ilelann I like the direction in which you're taking things. If this was a way we could provide a cross-platform `configure.sh`/`build.sh`, I would be in favor of scrapping them for a unified solution. Thanks for keeping up on this.

**Pull request merged**